### ISSUE TITLE
Video Capture Device does not capture iOS device audio

### ISSUE THREAD

**Issue by MisutaaUrufu** [obsproject/obs-studio#3661]
## Platform

Operating system and version: macOS 11.0 (20A5395g)
OBS Studio version: 26.0.2

## Expected Behavior

When adding an iOS device using a Video Capture source, ***audio output*** for the source should be added within the sound mixing panel of OBS.

## Current Behavior

- Audio is not captured nor controlled by OBS
- As OBS has activated A/V capture, system disables _audio-only pass-through_ for iOS device in **Audio MIDI Setup**.
- As *audio-only pass-through* is disabled, by design, OBS cannot capture the iOS device when adding an Audio Capture source.

## Steps to Reproduce

1. Create a scene.
2. Connect an unlocked, trusted iOS device.
3. Add a Video Capture source in OBS.
4. Choose the iOS device as the source device.
5. Add an Audio Capture source in OBS.
6. Fail to locate the iOS device from step 4 in the drop-down list.

## Workaround
1. Connect an unlocked, trusted iOS device.
2. Open QuickTime.
3. Open a New Movie Recording window.
4. Set the iOS device as both the video and audio capture device.
5. Pass system audio through a virtual audio device, such as Blackhole or Soundflower.
6. Add new Window Capture source in OBS.
7. Set QuickTime as the Window Capture window.
8. Set Desktop Audio device in settings to the virtual audio device from step 5.

## Additional information

macOS disables the separate pass-through audio device, for a connected iOS device, when a program begins A/V capture of said device. The capturing program can access both Audio & Video (e.g. QuickTime) capture, however. This suggests that the method of enumerating iOS audio devices when A/V capture is in progress is different from standard system audio devices.

**Comment by WizardCM** [comment]
I've definitely seen this issue in our support channels, but it did not occur to me that something on our side might not be handling a potential audio feed that could be coming through.

**Comment by MisutaaUrufu** [comment]
> I've definitely seen this issue in our support channels, but it did not occur to me that something on our side might not be handling a potential audio feed that could be coming through.

This issue actually existed in Catalina as well, and most forum posts I have found on the topic say it's "just" an OBS thing, and recommend similar workarounds as I have suggested, which capture the iOS device as part of quicktime, rather than directly in OBS.

If there's anything I can do to troubleshoot, let me know. It's definitely an OBS thing; How macOS handles the audio feed when capturing iOS device's I do not know, but I imagine [AVFoundation](https://developer.apple.com/documentation/avfoundation) is involved, seeing as it has facilities for both image preview and A/V capture, and is the system API for handling capture devices. — Perhaps OBS is strictly looking for audio devices on the system, and is not configured to accept [audio capture as part of a video capture from AVFoundation](https://developer.apple.com/documentation/avfoundation/avcaptureaudiodataoutput)?

**Comment by WizardCM** [comment]
Yep, we use AVFoundation. For anyone who wants to dig into this issue, [here's where the macOS Video Capture Device source](https://github.com/obsproject/obs-studio/blob/master/plugins/mac-avcapture/av-capture.mm) can be found. I notice that audio is not explicitly mentioned in the file, so you are correct that it's not configured to accept audio capture.

**Comment by Trinijoej** [comment]
After hours of trying to get the sound from my phone into OBS i decided to see if my handy Lighting to HDMI adapter would help solve this problem and I found the fix where you can record/stream audio video, capture video and audio from your IPHONE to OBS. I just so happen to have the IPHONE to HDMI adapter  and a USB to HDMI video capture and simple connected an HDMI cable to the the adapter out of the phone to the HDMI to USB adapter and selected the USB capture in OBS and it shows both audio and video through the HDMI connection. The IPHONE HDMI adapter is $29 to $50 from apple and the USB to HDMI Capture is $15 from Amazon.  THUS the workaround.

**Comment by MisutaaUrufu** [comment]
> The IPHONE HDMI adapter is $29 to $50 from apple and the USB to HDMI Capture is $15 from Amazon. THUS the workaround.

This isn't really a solution though. macOS exposes iOS devices as a native capture device. However, it exposes it as a *joint AV device*, compared to the separated audio and video devices of other capture cards.

Just like with eGPUs before it, OBS *should* be querying what is available, and surfacing relevant functionality. But instead, like older builds with eGPUs, OBS *assumes* that hardware will be surfaced in a particular way, and doesn't check if this is true.

In this case, it is essentially "throwing away" the audio of iOS devices capture natively. No one should have to spend money on something unnecessarily because of an oversight. :u

**Comment by gxalpha** [comment]
In case you haven't seen, there's a pull request open for this issue: #5449
It would be great to have some more testing on that, especially regarding long-term sync of video and audio, so feel free to test that.

**Comment by ArtemBernatskyy** [comment]
@gxalpha can test that if you will build https://github.com/obsproject/obs-studio/pull/5449 for M1.
Currently actively streaming using Macbook Pro 14 (2021) M1 32GB using unofficial OBS builds from github :(
